lib/teleterm/vnet/service.go

```diff
@@ -217,85 +214,29 @@ func (s *Service) Stop(ctx context.Context, req *api.StopRequest) (*api.StopResp
+// process hasn't configured a recent change yet.
 func (s *Service) ListDNSZones(ctx context.Context, req *api.ListDNSZonesRequest) (*api.ListDNSZonesResponse, error) {
 	// Acquire the lock just to check the status of the service. We don't want the actual process of
 	// listing DNS zones to block the user from performing other operations.
 	s.mu.Lock()
-
 	if s.status != statusRunning {
+		s.mu.Unlock()
 		return nil, trace.CompareFailed("VNet is not running")
 	}
+	osConfigProvider := s.vnetProcess.GetOSConfigProvider()
+	s.mu.Unlock()
 
-	defer s.mu.Unlock()
-
-	profileNames, err := s.cfg.DaemonService.ListProfileNames()
+	targetOSConfig, err := osConfigProvider.GetTargetOSConfiguration(ctx)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-
-	dnsZones, _ := s.listDNSZonesAndCIDRRanges(ctx, profileNames)
-
 	return &api.ListDNSZonesResponse{
-		DnsZones: dnsZones,
+		DnsZones: targetOSConfig.GetDnsZones(),
 	}, nil
 }
 
-func (s *Service) listDNSZonesAndCIDRRanges(ctx context.Context, profileNames []string) (dnsZones []string, cidrRanges []string) {
-	for _, profileName := range profileNames {
-		rootClusterURI := uri.NewClusterURI(profileName)
-		cLog := log.With("cluster", rootClusterURI)
-
-		rootClient, err := s.cfg.DaemonService.GetCachedClient(ctx, rootClusterURI)
-		if err != nil {
-			cLog.WarnContext(ctx, "Failed to create root cluster client, profile may be expired, skipping DNS zones of this cluster", "error", err)
-			continue
-		}
-		clusterConfig, err := s.clusterConfigCache.GetClusterConfig(ctx, rootClient)
-		if err != nil {
-			cLog.WarnContext(ctx, "Failed to load VNet configuration, profile may be expired, skipping DNS zones of this cluster", "error", err)
-			continue
-		}
-
-		dnsZones = append(dnsZones, clusterConfig.DNSZones...)
-		cidrRanges = append(cidrRanges, clusterConfig.IPv4CIDRRange)
-
-		leafClusters, err := s.cfg.DaemonService.ListLeafClusters(ctx, rootClusterURI.String())
-		if err != nil {
-			cLog.WarnContext(ctx, "Failed to list leaf clusters, profile may be expired, skipping DNS zones from leaf clusters of this cluster", "error", err)
-			continue
-		}
-
-		for _, leafCluster := range leafClusters {
-			cLog := log.With("cluster", leafCluster.URI.String())
-
-			clusterClient, err := s.cfg.DaemonService.GetCachedClient(ctx, leafCluster.URI)
-			if err != nil {
-				cLog.WarnContext(ctx, "Failed to create leaf cluster client, skipping DNS zones for this leaf cluster", "error", err)
-				continue
-			}
-			clusterConfig, err := s.clusterConfigCache.GetClusterConfig(ctx, clusterClient)
-			if err != nil {
-				cLog.WarnContext(ctx, "Failed to load VNet configuration, skipping DNS zones for this leaf cluster", "error", err)
-				continue
-			}
-
-			dnsZones = append(dnsZones, clusterConfig.DNSZones...)
-			cidrRanges = append(cidrRanges, clusterConfig.IPv4CIDRRange)
-		}
-	}
-
-	dnsZones = utils.Deduplicate(dnsZones)
-	cidrRanges = utils.Deduplicate(cidrRanges)
```

Comment:
I see this was dropped from the `LocalOSConfigProvider`, was this intentional?

Reply:
I think it needs to be added back, otherwise the diag report has duplicates of IPv4 CIDR range if each root cluster has the default one.

Reply:
oh I'll add it back, it's missing because I actually moved the LocalOSConfigProvider impl from localAppProvider which didn't have the dedupe